pkg/diagnostics/collectors.go

```diff
@@ -380,6 +380,56 @@ func (c *collectorFactory) crdCollector(crdType string) *Collect {
 	}
 }
 
+// APIServerCollector collect connection info when running a pod on an existing cluster
+func (c *collectorFactory) APIServerCollector(controlPlaneIP string) []*Collect {
+	var collectors []*Collect
+	collectors = append(collectors, c.generateRunPodsCollector(controlPlaneIP)...)
+	// collectors = append(collectors, c.pingLogCollectors()...)
+	return collectors
+}
+
+func (c *collectorFactory) generateRunPodsCollector(controlPlaneIP string) []*Collect {
+	return c.runPodsCollectors(controlPlaneIP)
+}
+
+func (c *collectorFactory) runPodsCollectors(controlPlaneIP string) []*Collect {
```

Comment:
Can I recommend that we break this up into two, separate generic collector methods that can be more easily reused in other use cases?

for example:

`pingHostCollector(hostIP string) *Collect`
`hostPortsCollector([]string ports, hostIP string) *Collect`

and then return these as part of a parent method, such as:

`controlPlaneNetworkPathCollector(controlPlaneIP string) []*Collect`

that way we can re-use the functionality for purposes other than just the CP IP availability.